packages/babel-plugin/src/utils/build-styled-component.ts

```diff
@@ -150,22 +183,16 @@ const getDestructuredProps = (node: t.Node): string[] => {
  *
  * @param path MemberExpression path
  */
-const handleMemberExpressionInStyledInterpolation = (path: NodePath<t.MemberExpression>) => {
+const handleMemberExpressionInStyledInterpolation = (
```

Comment:
I like @at-nathan idea to assume all props will have same identifier to simplify `props` so that we can simplify most of the logic, not only here. Eg I guess we can do the transformation before getting here

Reply:
To me it looks like we're adding on more logic to an already cluttered and non straight forward way of handling props and other dependencies. Should we rethink it and refactor to a cleaner solution?

Reply:
Is it something that we want to look at now or at a later stage?